Userland/Libraries/LibCore/Stream.cpp

```diff
@@ -90,14 +90,14 @@ ErrorOr<off_t> SeekableStream::size()
     return seek_result.value();
 }
 
-ErrorOr<File> File::open(StringView const& filename, OpenMode mode, mode_t permissions)
+ErrorOr<NonnullOwnPtr<File>> File::open(StringView const& filename, OpenMode mode, mode_t permissions)
 {
-    File file { mode };
-    TRY(file.open_path(filename, permissions));
+    auto file = TRY(adopt_nonnull_own_or_enomem(new File(mode)));
```

Comment:
If you want to use `adopt_nonnull_own_or_enomem`, you should use `new (nothrow)`